src/uu/mv/src/mv.rs

```diff
@@ -485,7 +485,10 @@ fn rename(
             if is_empty_dir(to) {
                 fs::remove_dir(to)?;
             } else {
-                return Err(io::Error::new(io::ErrorKind::Other, "Directory not empty"));
+                return Err(io::Error::new(
+                    io::ErrorKind::Other,
+                    "A directory with the same name exists at destination",
+                ));
```

Comment:
Sorry, but I don't think this correct. The check that we are doing in `is_empty_dir` is not about the same directory already existing. So this message would actually be wrong in a lot of cases. The solution here needs to be a bit more complex than changing the behaviour in all cases.

Reply:
Do you mean that this could error could be shown in instances other than the use described in the issue or test?

I definitely do agree that the default GNU message of simply "directory not empty" is confusing for those not familiar. Did you have an idea for something that better fits the behavior?